src/Bicep.Core/TypeSystem/TypeValidator.cs

```diff
@@ -186,6 +186,11 @@ private TypeSymbol NarrowType(TypeValidatorConfig config, SyntaxBase expression,
             {
                 // certain properties such as scope, parent, dependsOn do not allow values of "any" type
                 diagnosticWriter.Write(config.OriginSyntax ?? expression, x => x.AnyTypeIsNotAllowed());
+
+                // if we let the type narrowing continue, we could get more diagnostics
+                // but it also leads to duplicate "disallow any" diagnostics caused by the union type narrowing
+                // (occurs with "dependsOn: [ any(true) ]")
+                return targetType;
```

Comment:
>targetType [](http://example.com/codeflow?start=23&length=10)

This is a fix for a pre-existing issue that was exposed by union type member type narrowing that was added in this PR.